py/kubeflow/testing/cd/update_kf_apps.py

```diff
@@ -229,6 +338,77 @@ def create_runs(config, output_dir, template, src_dir):
       logging.info("Succcessfully created pipeline runs for all apps and "
                    "versions")
 
+    logging.info(f"Pipelines that need to be run {len(pipelines_to_run)} of "
+                 f"{len(all_pipelines)}")
+    logging.info("The pipelines that need to be run are:\n%s","\n".join(
+      pipelines_to_run))
+
+    return all_pipelines, pipelines_to_run
+
+  @staticmethod
+  def apply(config, output_dir, template, src_dir, namespace):
+    """Create PipelineRuns for any applications that need to be updated."""
+
+    k8s_config.load_kube_config(persist_config=False)
+
+    client = k8s_client.ApiClient()
+    crd_api = k8s_client.CustomObjectsApi(client)
+
+    _, pipelines_to_run = UpdateKfApps.create_runs(
+      config, output_dir, template, src_dir)
+
+    if not pipelines_to_run:
+      logging.info("No pipelines need to be run")
+      return
+
+    #pipelines_to_run = ["/tmp/runs/admission-webhook-run-master-6b987df8.yaml",
+                        #"/tmp/runs/kfam-run-master-caf5eb33.yaml"]
+    for p in pipelines_to_run:
+      with open(p) as hf:
+        run = yaml.load(hf)
+
+      group, version = run["apiVersion"].split("/", 1)
+      kind = run["kind"]
+      plural = kind.lower() + "s"
```

Comment:
I'm wondering if this is a robust way to get kind plural, although it should work in most cases.

Reply:
Since its a custom resource in this case we could do `kubectl get crd` and the plurals is defined in the CRD. At this point though its not really necessary since we really only work with tekton pipeline runs which we know the heuristic works for.

I'm not sure what you would do for built in objects; e.g. Deployment since there is no CRD.

Reply:
ah, I missed they are tekton runs but was thinking of all possible kinds.